integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java

```diff
@@ -41,42 +46,62 @@ public void ingressOfBridgeIsAvailableWithinMinutes(String testBridgeName, int t
     public void ingressOfBridgeIsNotAvailableWithinMinutes(String testBridgeName, int timeoutMinutes) {
         String endpoint = BridgeUtils.getOrRetrieveBridgeEndpoint(context, testBridgeName);
 
-        Awaitility.await().atMost(Duration.ofMinutes(timeoutMinutes)).pollInterval(Duration.ofSeconds(5))
+        Awaitility.await()
+                .conditionEvaluationListener(new AwaitilityOnTimeOutLogger(
+                        () -> IngressResource.optionsJsonEmptyEventResponse(context.getManagerToken(), endpoint)))
+                .atMost(Duration.ofMinutes(timeoutMinutes)).pollInterval(Duration.ofSeconds(5))
                 .untilAsserted(() -> IngressResource.optionsJsonEmptyEventResponse(context.getManagerToken(), endpoint)
                         .then()
                         .statusCode(Matchers.anyOf(Matchers.is(404), Matchers.is(503))));
     }
 
-    @Then("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\":$")
-    public void sendCloudEventToIngressOfBridge(String testBridgeName, String cloudEvent) {
+    @When("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\" with path \"([^\"]*)\":$")
+    public void sendCloudEventToIngressOfBridgeWithPath(String testBridgeName, String path, String cloudEvent) {
+        sendAndCheckCloudEvent(testBridgeName, cloudEvent, path, 200);
+    }
+
+    @When("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\" with path \"([^\"]*)\" and default headers:$")
+    public void sendPlainCloudEventToIngressOfBridgeWithPath(String testBridgeName, String path, String cloudEvent) {
+        sendAndCheckCloudEvent(testBridgeName, cloudEvent, path, getDefaultCloudEventHeaders(), 200);
+    }
+
+    @When("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\" with path \"([^\"]*)\" is failing with HTTP response code (\\d+):$")
+    public void sendCloudEventToIngressOfBridgeWithPathIsFailingWithHTTPResponseCode(String testBridgeName, String path,
+            int responseCode, String cloudEvent) {
+        sendAndCheckCloudEvent(testBridgeName, cloudEvent, path, responseCode);
+    }
+
+    @When("^send a cloud event to the Ingress of the Bridge \"([^\"]*)\" with path \"([^\"]*)\" and default headers is failing with HTTP response code (\\d+):$")
+    public void sendPlainCloudEventToIngressOfBridgeWithPathIsFailingWithHTTPResponseCode(String testBridgeName,
+            String path, int responseCode, String cloudEvent) {
+        sendAndCheckCloudEvent(testBridgeName, cloudEvent, path, getDefaultCloudEventHeaders(), responseCode);
+    }
```

Comment:
please revisit the method names so they match steps description better

Reply:
from another PR but will correct it